### PR TITLE
compiler: Fix zero to zero slices

### LINKED ISSUE #1756: f.data[0:0] != np.array(f.data)[0:0]
Reproducer

```
In [8]: from devito import Grid, Function                                                                                                                                                                          

In [9]: grid = Grid(shape=(4, 4))                                                                                                                                                                                  

In [10]: f = Function(name='f', grid=grid)                                                                                                                                                                         

In [11]: f.data                                                                                                                                                                                                    
Allocating memory for f(6, 6)
Out[11]: 
Data([[0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.]], dtype=float32)

In [12]: f.data[0:0]                                                                                                                                                                                               
Out[12]: 
Data([[0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.]], dtype=float32)

In [13]: f.data[slice(0, 0)]                                                                                                                                                                                       
Out[13]: 
Data([[0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.]], dtype=float32)

In [14]: np.array(f.data)                                                                                                                                                                                          
Out[14]: 
array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]], dtype=float32)

In [15]: np.array(f.data)[0:0]                                                                                                                                                                                     
Out[15]: array([], shape=(0, 4), dtype=float32)

In [16]: np.array(f.data)[slice(0, 0)]                                                                                                                                                                             
Out[16]: array([], shape=(0, 4), dtype=float32)
```

issue cropping up at In [12], as f.data[0:0] should return an empty array
also note that the problem only seems to be 0:0 :

```
In [19]: f.data[0:0]                                                                                                                                                                                               
Out[19]: 
Data([[0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.]], dtype=float32)

In [20]: f.data[1:1]                                                                                                                                                                                               
Out[20]: Data([], shape=(0, 4), dtype=float32)

In [21]: f.data[3:3]                                                                                                                                                                                               
Out[21]: Data([], shape=(0, 4), dtype=float32)

In [22]: f.data[-1:-1]                                                                                                                                                                                             
Out[22]: Data([], shape=(0, 4), dtype=float32)

In [23]: f.data[:, 0:0]                                                                                                                                                                                            
Out[23]: 
Data([[0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.],
      [0., 0., 0., 0.]], dtype=float32)

In [24]: f.data[:, -1:-1]                                                                                                                                                                                          
Out[24]: Data([], shape=(4, 0), dtype=float32)

In [25]: f.data[:, 2:2]                                                                                                                                                                                            
Out[25]: Data([], shape=(4, 0), dtype=float32)
15:41
(in any dimension)
```

### PR DESCRIPTION
**Pull request by rhodrin**
Fixes a bug where slices of `Function`'s of the form `f.data[0:0]` were returning all data instead of an empty array of shape `(0, n, ...)`.

Fixes #1756.

### CODE REVIEW

File: devito/builtins/utils.py [hunk 1]
```diff
@@ -61,7 +61,13 @@ def nbl_to_padsize(nbl, ndim):
     else:
         raise ValueError("`nbl` must be an integer or tuple of (tuple of) integers"
                          "of length `function.ndim`.")
-    return tuple(nb_pad)
+    pad_slice = []
```
Comment (FabioLuporini) [inline devito/builtins/utils.py:64]: aka 

```
pad_slices = [(i, None if j == 0 else -j) for i, j in nb_pad]
```

?

**Review by FabioLuporini — COMMENTED** [review]
commit messages should follow the convention here: https://github.com/devitocodes/devito/wiki/Tags-for-commit-messages-and-PR-titles

but great ! thanks a lot !

**Comment by FabioLuporini** [comment]
since you are to rebase this one anyway, @rhodrin , note that the tag `data` you've used in the commit messages is not a legal tag. We use `compiler` as an umbrella term for anything touching stuff below the API

**Comment by FabioLuporini** [comment]
or... we add a new tag ;)

**Comment by rhodrin** [comment]
> or... we add a new tag ;)

Ta. Will fix when rebasing.

**Review by mloubout — APPROVED** [review]

**Review by FabioLuporini — APPROVED** [review]
LGTM

**Pull request merged**